docker-compose.yml

```diff
@@ -22,7 +29,16 @@ services:
       target: development
     environment:
       # Be sure to add environment variables to config/options.yml
-      - DATABASE_URL=postgres://app:secret@postgres/app_development
+      - HOSTEDGPT_DB_USERNAME=${HOSTEDGPT_DB_USERNAME:-app}
+      - HOSTEDGPT_DB_PASSWORD=${HOSTEDGPT_DB_PASSWORD:-secret}
+      - HOSTEDGPT_DB_HOST=${HOSTEDGPT_DB_HOST:-postgres}
+      - HOSTEDGPT_DB_PORT=${HOSTEDGPT_DB_PORT:-5432}
+      - HOSTEDGPT_DB_NAME=${HOSTEDGPT_DB_NAME:-app_development}
+      - DATABASE_URL
```

Comment:
When I look at this docker-compose, it still looks like we're setting things twice. the username "app" and password "secret" are being set on HOSTED_DB_USERNAME and also POSTGRES_USER, but isn't that what you were eliminating?

I know this whole change ended up being a bigger thing than you intended. Maintaining support for multiple docker configs & the non-docker config is really quite complicated.

Reply:
yeah support for the database url really added a lot. The postgres user is created in the init before our new init script is run. So we have to have that set as well and I'm not sure there's a way to take it from the database_url since the built-in init script runs before we can run any custom code